app/views/govuk_publishing_components/components/_layout_header.html.erb

```diff
@@ -13,63 +15,49 @@
-            ></path>
-            <image src="<%= asset_path('govuk-logotype-crown.png') %>" class="govuk-header__logotype-crown-fallback-image"></image>
-          </svg>
-          <span class="govuk-header__logotype-text">
-            GOV.UK
-          </span>
-        </span>
-        <% unless public_environment %>
-          <span class="govuk-header__product-name gem-c-header__product-name">
-            <%= product_name %>
-          </span>
-          <span class="gem-c-environment-tag govuk-tag gem-c-environment-tag--<%= environment %>">
-            <%= environment %>
-          </span>
-        <% end %>
-      </a>
-    </div><div class="govuk-header__content gem-c-header__content">
-    <% if navigation_items.any? %>
-    <button role="button" class="govuk-header__menu-button gem-c-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
-    <nav class="gem-c-header__nav">
-      <ul id="navigation" class="govuk-header__navigation govuk-header__navigation--end" aria-label="Top Level Navigation">
-        <% navigation_items.each_with_index do |item, index| %>
-          <li class="govuk-header__navigation-item <%= "govuk-header__navigation-item--active" if item[:active] %>
-            <%= "govuk-header__navigation-item--collapsed-menu-only" if item[:show_only_in_collapsed_menu] %>">
-            <%= link_to(
-              item[:text],
-              item[:href],
-              class: 'govuk-header__link',
-              ) %>
-          </li>
-        <% end %>
-      </ul>
-    </nav>
-    <% end %>
-
-    </div>
-    </div>
-
-    <% if search %>
-        <div class="govuk-grid-column-one-third gem-c-layout-header__search">
+<% if search_left %>
```

Comment:
What's the reasoning behind this `if` statement? It looks like there's quite a bit of shared code between the `if` and the `else` - would it not be possible to conditionally add the `gem-c-layout-header--search-left` to the existing HTML and work from there?

Reply:
It's because the div structure needs to be different with the nav rendered last if the search is on the left, otherwise the menu button for the mobile nav is hard to place properly. Looking at it, though, I can move more of it outside the if/else to reduce the repetition.